src/linker/Linker/OverrideInformation.cs

```diff
@@ -4,17 +4,17 @@
 namespace Mono.Linker {
 	[DebuggerDisplay ("{Override}")]
 	public class OverrideInformation {
-		public readonly MethodDefinition Base;
-		public readonly MethodDefinition Override;
-		public readonly InterfaceImplementation MatchingInterfaceImplementation;
-
 		public OverrideInformation (MethodDefinition @base, MethodDefinition @override, InterfaceImplementation matchingInterfaceImplementation = null)
 		{
 			Base = @base;
 			Override = @override;
 			MatchingInterfaceImplementation = matchingInterfaceImplementation;
 		}
 
+		public MethodDefinition Base { get; private set; }
+		public MethodDefinition Override { get; private set; }
+		public InterfaceImplementation MatchingInterfaceImplementation { get; private set; }
```

Comment:
Why add setters? It looks like we can keep these readonly since they're only set in the ctor.

Reply:
It's more resilient, one can add or change set/get behaviour if needed without breaking the contract.

Reply:
Adding getters makes sense to me - I am specifically asking about the setters. If you only add a getter, the underlying field can still be readonly without changing the contract, and you can always introduce setters later if needed: https://sharplab.io/#v2:C4LglgNgNAJiDUAfAAgJgIwFgBQyDMABGgQMIEDeOB1RhARgPYMQEBCFBA5gKbADcBAL5Ua+UgAoAlBRE0a7ALwFgAJwCu3PrKE5BQA=

Reply:
I see what you mean